BenchmarkVCFs/SimpleBenchmark.wdl

```diff
@@ -0,0 +1,592 @@
+        ln -s ~{input_vcf} input.vcf.gz
+        ln -s ~{input_vcf_index} input.vcf.gz.tbi
+
+        # Subset to given sample / loci; Output subsetted VCF
+        # Add variable padding to ensure capture of variants on boundaries downstream
+        # Remove unused alternates to make bcftools stats more accurate on multisample callsets
+        gatk SelectVariants \
+            -V input.vcf.gz \
+            -R ~{reference.fasta} \
+            ~{"-sn " + input_sample_name} \
+            ~{"-L " + stratifier.interval} \
+            ~{"--interval-padding " + interval_padding} \
+            --remove-unused-alternates \
+            -O subset.vcf.gz
+
+    >>>
+
+    runtime{
+        docker: "us.gcr.io/broad-gatk/gatk:" + gatk_tag
+        preemptible: select_first([preemptible, 0])
+        disks: "local-disk " + disk_size + " HDD"
+        bootDiskSizeGb: "16"
+        cpu: cpu
+        memory: memory + " GB"
+    }
+
+    output {
+        File output_vcf = "subset.vcf.gz"
+        File output_vcf_index = "subset.vcf.gz.tbi"
+    }
+}
+
+task VCFEval {
+    input {
+        # Input VCF Files
+        File call_vcf
+        File call_vcf_index
+        String call_output_sample_name
+        File base_vcf
+        File base_vcf_index
+        String base_output_sample_name
+
+        Reference reference
+
+        # Interval File to Subset Analysis on given actual truth data
+        File? evaluation_bed
+
+        # String for VCF field to use as ROC score
+        String score_field
+
+        # Label to add to ROC table outputs
+        String strat_label
+
+        # vcfeval Arguments
+        Boolean passing_only = true
+        Boolean require_matching_genotypes = true
+        Boolean enable_ref_overlap = false
+
+        # Runtime params
+        Int? preemptible
+        Int disk_size = ceil(size(call_vcf, "GB") + size(base_vcf, "GB") + size(reference.fasta, "GB")) + 25
+        Int cpu = 8
+        Int memory = 16
+    }
+
+    command <<<
+        set -xeuo pipefail
+
+        rtg format -o rtg_ref ~{reference.fasta}
+        rtg vcfeval \
+            ~{false="--all-records" true="" passing_only} \
+            ~{false="--squash-ploidy" true="" require_matching_genotypes} \
+            ~{true="--ref-overlap" false="" enable_ref_overlap} \
+            -b ~{base_vcf} \
+            -c ~{call_vcf} \
+            ~{"-e " + evaluation_bed} \
+            --vcf-score-field="~{score_field}" \
+            --output-mode split \
+            --decompose \
+            --roc-subset snp,mnp,indel \
+            -t rtg_ref \
+            -o output_dir \
+
+        python3 << CODE
+        import gzip
+        import pandas as pd
+
+        roc_summary = pd.DataFrame()
+        for Type in ['snp', 'mnp', 'indel']:
+            file_path = f'output_dir/{Type}_roc.tsv.gz'
+            with gzip.open(file_path, 'r') as file:
+                roc = [line.decode('utf-8') for line in file.readlines()]
```

Comment:
Despite the compression, these files are usually very small, e.g. less than 100-1000 lines total. I think your first suggestion would still involve loading the full file to do the list comprehension. Also I think `'rt'` mode is equivalent to `'r'` since `'t'` is a default. The while loop should work, but feels like overkill to avoid reading 100 lines or so.

Reply:
As you mentioned, the default for gzip.open is `'rb'` rather than `'rt'` so I changed this and took out the decode step.

Reply:
You're still saving the entire file into memory (which you said isn't a big deal, but still unnecessary). You can just in the line below write `for line in file` instead of `for line in roc`

Reply:
Good point. I think I updated it so now it'll just read the first few lines of the file, and then stop once it hits a non-header line (i.e. one not starting with a `#`).